src/test/java/nearsoft/academy/bigdata/recommendation/MovieRecommender.java

```diff
@@ -0,0 +1,140 @@
+
+
+    public MovieRecommender(String filePath) {
+        this.users = new HashMap<String, Integer>();
+        this.products = HashBiMap.create();
+        File csvFile = generateCSV(filePath);
+        buildRecommender(csvFile);
+    }
+
+    private File generateCSV(String filePath) {
+        try {
+            BufferedReader reader = getGzipReader(filePath);
+            FileWriter writer = new FileWriter(OUTPUT_FILEPATH);
+
+            String currentLine;
+            String csvLine = "";
+            int currentProduct = 0;
+
+            while((currentLine = reader.readLine()) != null) {
+                if (currentLine.startsWith(PRODUCT_KEY)) {
+                    String productId = currentLine.substring(19);
+                    if (!this.products.containsKey(productId))
+                        this.products.put(productId, this.products.size());
+
+                    currentProduct = this.products.get(productId);
+                }
+
+                else if (currentLine.startsWith(USER_KEY)) {
+                    String userId = currentLine.substring(15);
+                    if (!this.users.containsKey(userId))
+                        this.users.put(userId, users.size());
+                    this.totalReviews++;
+
+                    csvLine = this.users.get(userId) + "," + currentProduct + ",";
+                }
+
+                else if (currentLine.startsWith(SCORE_KEY)) {
+                    double score = Double.parseDouble(currentLine.substring(14));
+                    csvLine += score + "\n";
+                    writer.write(csvLine);
+                    writer.flush();
+                }
+            }
+
+            reader.close();
+            writer.close();
+
+        } catch (IOException ex) {
+            ex.printStackTrace();
+        }
+
+        return new File(OUTPUT_FILEPATH);
+    }
+
+    public void buildRecommender(File csvFile) {
+        try {
+            DataModel model = new FileDataModel(csvFile);
+            UserSimilarity similarity = new PearsonCorrelationSimilarity(model);
+            UserNeighborhood neighborhood = new ThresholdUserNeighborhood(0.1, similarity, model);
+            this.recommender = new GenericUserBasedRecommender(model, neighborhood, similarity);
+
+        } catch (Exception ex) {
+            ex.printStackTrace();
+        }
+    }
+
+    public List<String> getRecommendationsForUser(String userId)  {
+        try {
+            int userIdInt = this.users.get(userId);
+            List recommendations = this.recommender.recommend(userIdInt, 5);
+            return getRecommendationsIds(recommendations);
+        } catch (TasteException ex) {
+            ex.printStackTrace();
+        }
+
+        return null;
+    }
+
+    private List<String> getRecommendationsIds(List<RecommendedItem> recommendations) {
+        BiMap<Integer, String> invertedProducts = this.products.inverse();
+        ArrayList<String> productIds = new ArrayList<String>();
```

Comment:
Also, since the principal responsibility of that list is adding new elements, I would recommend to use LinkedList instead: http://stackoverflow.com/questions/322715/when-to-use-linkedlist-over-arraylist